### PR TITLE
shutter: 0.94.3 -> 0.97 

### LINKED ISSUE #132324: Shutter fails to build.
### Describe the bug
Shutter fails to build, looks like an issue with Imagemagick.

### Steps To Reproduce
1. `nix-shell -p shutter`

### Expected behavior
Expect that it should build, it was working on my last generation.

### Notify maintainers

@bjornfor
### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.4, NixOS, 21.05.1970.11c662074e2 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.05.1970.11c662074e2, home-manager-21.05, nixos-unstable-21.11pre304061.dd14e5d78e9"`
 - channels(yisraeldov): `"nixos-21.05.1669.973910f5c31"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`


```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```
Probably related to #132237

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Closes #132324

###### Things done
i tested the deps manually one by one

Result of `nixpkgs-review pr 132565` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>gscan2pdf</li>
    <li>perl532Packages.GooCanvas2CairoTypes</li>
    <li>perl532Packages.Gtk3ImageView</li>
    <li>perl534Packages.GooCanvas2CairoTypes</li>
    <li>perl534Packages.Gtk3ImageView</li>
    <li>shutter</li>
  </ul>
</details>

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/shutter/default.nix [hunk 1]
```diff
@@ -1,39 +1,104 @@
...
+, gdk-pixbuf
+, librsvg
+, hicolor-icon-theme
+, procps
+, libwnck
+, libappindicator-gtk3
 }:
 
 let
   perlModules = with perlPackages;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/shutter/default.nix:16]:
```suggestion
  perlModules = with perlPackages; [
```

File: pkgs/applications/graphics/shutter/default.nix [hunk 2]
```diff
@@ -1,39 +1,104 @@
...
-      XMLParser HTTPMessage ProcSimple SortNaturally LocaleGettext
-      ProcProcessTable URI ImageExifTool Gtk2AppIndicator LWP JSON
-      ImageMagick WWWMechanize HTTPDate HTMLForm HTMLParser HTMLTagset JSONMaybeXS
-      commonsense HTTPCookies NetOAuth PathClass GooCanvas X11Protocol Cairo
-      EncodeLocale TryTiny TypesSerialiser LWPMediaTypes
+    [
+      # Not sure if these are needed
+      # Gnome2 Gnome2Canvas Gnome2VFS Gtk2AppIndicator Gtk2Unique
+      ImageMagick
+      Cairo
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/shutter/default.nix:21]:
```suggestion
    # Not sure if these are needed
    # Gnome2 Gnome2Canvas Gnome2VFS Gtk2AppIndicator Gtk2Unique
    ImageMagick
    Cairo
```

File: pkgs/applications/graphics/shutter/default.nix [hunk 3]
```diff
@@ -1,39 +1,104 @@
...
+    "prefix=${placeholder "out"}"
+  ];
 
-    wrapProgram $out/bin/shutter \
-      --set PERL5LIB "${perlPackages.makePerlPath perlModules}" \
-      --prefix PATH : "${imagemagick.out}/bin" \
-      --suffix XDG_DATA_DIRS : "${hicolor-icon-theme}/share" \
-      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE"
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/shutter/default.nix:92]: What exactly needs patching?

File: pkgs/applications/graphics/shutter/default.nix [hunk 4]
```diff
@@ -1,39 +1,104 @@
...
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --set PERL5LIB ${perlPackages.makePerlPath perlModules} \
+      --prefix PATH : ${imagemagick.out}/bin \
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/shutter/default.nix:98]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ imagemagick.out]} \
```

File: pkgs/top-level/perl-packages.nix [hunk 5]
```diff
@@ -9008,6 +9008,20 @@ let
     };
   };
 
+  GooCanvas2CairoTypes = buildPerlPackage {
+    pname = "GooCanvas2-CairoTypes";
+    version = "0.001";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/A/AS/ASOKOLOV/GooCanvas2-CairoTypes-0.001.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:9015]:
```suggestion
  GooCanvas2CairoTypes = buildPerlPackage rec {
    pname = "GooCanvas2-CairoTypes";
    version = "0.001";
    src = fetchurl {
      url = "mirror://cpan/authors/id/A/AS/ASOKOLOV/GooCanvas2-CairoTypes-${version}.tar.gz";
```

File: pkgs/applications/graphics/shutter/default.nix (hunk 3 above)
Comment (Artturin) [reply pkgs/applications/graphics/shutter/default.nix:92]: https://github.com/shutter-project/shutter/blob/master/po2mo.sh
made it explicit

**Comment by Artturin** [comment]
Thanks

**Comment by Artturin** [comment]
weird i pushed to my branch but its not showing here https://github.com/Artturin/nixpkgs/tree/shutter

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132565` at 5e7caf78 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>gscan2pdf</li>
    <li>perl532Packages.GooCanvas2CairoTypes</li>
    <li>perl532Packages.Gtk3ImageView</li>
    <li>perl534Packages.GooCanvas2CairoTypes</li>
    <li>perl534Packages.Gtk3ImageView</li>
    <li>shutter</li>
  </ul>
</details>

**Pull request merged**